Packet++/header/HttpLayer.h

```diff
@@ -207,201 +207,217 @@ namespace pcpp
 		HttpRequestFirstLine* m_FirstLine;
 	};
 
-
-
-
-
-	// -------- Class HttpResponseLayer -----------------
-
-	class HttpResponseFirstLine;
-
+	// -------- Class HttpResponseStatusCode -----------------
 
 	/**
-	 * @class HttpResponseLayer
-	 * Represents an HTTP response header and inherits all basic functionality of HttpMessage and TextBasedProtocolMessage.
-	 * The functionality that is added for this class is the HTTP first line concept. An HTTP response has the following first line:
-	 * <i>200 OK HTTP/1.1</i>
-	 * Since it's not an "ordinary" HTTP field, it requires a special treatment and gets a class of it's own: HttpResponseFirstLine.
-	 * Unlike most L2-4 protocols, an HTTP response header can spread over more than 1 packet. PcapPlusPlus currently doesn't support a header
-	 * that is spread over more than 1 packet so in such cases: 1) only the first packet will be parsed as HttpResponseLayer (the other packets
-	 * won't be recognized as HttpResponseLayer) and 2) the HTTP header for the first packet won't be complete (as it continues in the following
-	 * packets), this why PcapPlusPlus can indicate that HTTP response header is complete or not (doesn't end with "\r\n\r\n" or "\n\n") using
-	 * HttpMessage#isHeaderComplete()
+	 * Class for the enum of HTTP response status codes
 	 */
-	class HttpResponseLayer : public HttpMessage
+	class HttpResponseStatusCode
```

Comment:
Ideally we should strive to keep backward compatibility as much as possible, meaning users' code should break once they build their application with this new code.

In order to do this we can:
- Keep `HttpResponseStatusCode` a sub-class of `HttpResponseLayer`
- Keep all the enum values that were previously supported
- Maybe implement the cast to int operator so it's easier to do `static_cast<int>(value)`? 🤔

Reply:
1. okay
2. okay, but what is the code meaning that has changed in the new spec? 
3. it's already supported, currently, it's fine to call `static_cast<int>(HttpResponseStatusCode::Http417ExpectationFailed)` (get `417`)

Reply:
> 2\. okay, but what if the code meaning that has changed in the new spec?

Let's discuss specific use-cases, hopefully there aren't too many...



> 3\. it's already supported, currently, it's fine to call `static_cast<int>(HttpResponseStatusCode::Http417ExpectationFailed)` (get `417`)

Awesome! 👍

Reply:
1. done
2. done
3. done